docs: DVP update due v1.10.0 (#21862)
deckhouse/deckhouse: docs/site/pages/virtualization-platform/documentation/user/resource-managment/VM_POOLS.md

        ttl: 30m
      spec:
        persistentVolumeClaim:
          size: 50Gi
EOF
```

Replicas are named `<pool>-<random>`. Disks follow the same scheme: a per-replica (`Delete`) disk is named `<replica>-<template>` (for example `runners-1b2e84-root`), a reusable (`Retain`) disk is named `<pool>-<template>-<random>`. List replicas with `d8 k get vm -l vmpool.virtualization.deckhouse.io/pool=runners`.

## Attaching a shared CD-ROM (or any shared image) to every replica

Besides the per-replica disks, `blockDeviceRefs` may reference read-only images — a `ClusterVirtualImage` or a `VirtualImage`. Such an image is not per-replica: every replica attaches the same image (for example a common ISO with tools or drivers). Images are not listed in `virtualDiskTemplates` (they have no per-replica state) and are not subject to the bijection.

Add the image to `blockDeviceRefs` at the position where you want it in the boot order — for an OS-installation ISO, place it before the disk; for a tools CD-ROM, after:

```yaml
spec:
  virtualMachineTemplate:
    spec:
      blockDeviceRefs:
        - kind: VirtualDisk           # per-replica writable root, boots first
          name: root

>         - kind: ClusterVirtualImage   # shared read-only CD-ROM, attached to every replica

Reviewer: ```suggestion
        - kind: VirtualDisk           # Per-replica writable root, boots first.
          name: root
        - kind: ClusterVirtualImage   # Shared read-only CD-ROM, attached to every replica.
```

    name: runners
  minReplicas: 3
  maxReplicas: 20
  metrics:
    - type: Resource
      resource:
        name: cpu
        target:
          type: Utilization
          averageUtilization: 70
```

Beyond CPU/memory, the pool also works with custom metrics (`Pods`/`External` via `custom.metrics.k8s.io`/`external.metrics.k8s.io`) and KEDA, for example to scale on an external queue length. With `scaleDownPolicy: Explicit` an autoscaler can only scale up: anonymous scale-down through the `scale` subresource is rejected (remove replicas by name, see below).

The `spec.scaleDownPolicy` field selects which replica is removed on anonymous scale-down:

- `NewestFirst`: the youngest replicas are removed first.
- `OldestFirst`: the oldest replicas are removed first.
- `Explicit`: anonymous scale-down is rejected; replicas can be removed only by name (see below). Use it when only the caller knows which replica is safe to remove (for example, an idle one).

## Removing specific replicas

By default, when the pool shrinks the controller chooses which replica to remove.

To remove particular replicas (and shrink the pool by that count), use the `scaleDownWith` subresource:

```bash

> kubectl create --raw \

Reviewer: `kubectl` -> `d8 k`?

Author: ```suggestion
d8 k create --raw \
```

  /apis/subresources.virtualization.deckhouse.io/v1alpha2/namespaces/ci/virtualmachinepools/runners/scaledownwith \
  -f - <<'EOF'
{"targets": ["runners-1b2e84", "runners-9c0d11"]}
EOF
```

> A plain `kubectl delete vm` does not shrink the pool: the controller treats it as a lost replica and creates a replacement.

Author: ```suggestion
A plain `d8 k delete vm` does not shrink the pool: the controller treats it as a lost replica and creates a replacement.
```
